packages/page-claims/src/index.tsx

```diff
@@ -73,75 +64,134 @@ const Signature = styled.textarea`
   }
 `;
 
-// FIXME Not React.FC yet
-class ClaimsApp extends TxModal<Props, State> {
-  constructor (props: Props) {
-    super(props);
-
-    this.defaultState = {
-      ...this.defaultState,
-      claim: null,
-      didCopy: false,
-      ethereumAddress: null,
-      signature: null,
-      step: 0
-    };
-    this.state = this.defaultState;
-  }
+function ClaimsApp (): React.ReactElement {
+  const [didCopy, setDidCopy] = useState(false);
+  const [ethereumAddress, setEthereumAddress] = useState<EthereumAddress | null>(null);
+  const [signature, setSignature] = useState<EcdsaSignature | null>(null);
+  const [step, setStep] = useState<Step>(Step.Account);
+  const [accountId, setAccountId] = useState<string | null>(null);
+  const { api, systemChain } = useApi();
+  const { t } = useTranslation();
+  const isPreclaimed = useCall<boolean>(api.query.claims.preclaims, [accountId], {
+    transform: (option: Option<EthereumAddress>) => {
+      if (option.isSome) {
+        setEthereumAddress(option.unwrap());
```

Comment:
Not convinced about setting the state inside the transform.

Reply:
I moved it out.